src/app/api/controllers/userController.js

```diff
@@ -23,11 +63,15 @@ const createUser = async (req, res) => {
 
   try {
     await userService.createUser(username, email, password);
-    return res.sendStatus(201);
+    let token = makeToken();
+    console.log("Generated token", token);
+    tokenStorage[token] = username.toUpperCase();
+    return res.cookie("token", token, cookieOptions).send();
   } catch (error) {
     switch (error.message) {
       case "Username or email already exists":
-        return res.status(409).send("Username or email already exists");
+        return res.status(400).send("Username or email already exists");
```

Comment:
409 is the correct error code here, it denotes a conflict, in the case the resource (the user with that username or email) already exists